File: Tasks/ip.js
"use strict";

// Splits ip
const parseIP = (input) => {
  if (!input || typeof input !== "string") return;

  const parts = input.split(".");
  if (parts.length !== 4) return;

  const result = parts.map((part) => {
    const num = parseInt(part, 10);
    if (isNaN(num) || num < 0 || num > 255 || part === "") return;
    return num;
  });

  if (result.includes(undefined)) return;

  return result;
};

module.exports = parseIP;

File: JavaTasks/src/main/java/com/exams/tasks/Ip.java
package com.exams.tasks;

public class Ip {
    public static int[] parse(String ip) {
        if (ip.isEmpty()) return null;

        String[] parts = ip.split("\\.");
        if (parts.length != 4) return null;

        int[] result = new int[4];
        for (int i = 0; i < parts.length; i++) {
            try {
                result[i] = Integer.parseInt(parts[i]);
                if (result[i] < 0 || result[i] > 255) {
                    return null;
                }
            } catch (NumberFormatException e) {
                return null;
            }
        }
        return result;
    }
}

